File: java-main/java-main/IO/Nestedfor5/p7.java
/*write a program to print the following pattern
Row =5;
O
14 13
L K J
9 8 7 6
E D C B A

Row = 4
10
I H
7 6 5
D C B A
*/

import java.util.Scanner;
class Demo{
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the number of rows: ");
        int row = input.nextInt();
        int num = 0;
        char ch = 'O';

        for(int i = 1; i <= row; i++) {
            for(int j = 1; j <= i; j++) {
                if(i % 2 != 0) {
                    System.out.print(ch + " ");
                    ch++;
                }
                else {
                    System.out.print(num + " ");
                    num--;
                }
            }
            if(i % 2 != 0) {
                ch--;
            }
            else {
                num += i + 4;
            }
            System.out.println();
        }
    }
}

File: bootcamp_c_cpp_dsa-main/bootcamp_c_cpp_dsa-main/C/pattern3/p7.c
#include<stdio.h>
void main(){

	int rows;
	printf("Enter num of rows:");
	scanf("%d",&rows);

	int i=1,j=2,l=3;
	int num=1;
	int f=0;

	for(int k=1;k<=rows;k++){
		for(int s=1;s<=rows;s++){
			if(i==num){
				printf("=\t");
				f=1;
			}else if(j==num){
				printf("s\t");
				f=2;
			}else if(l==num){
				printf("@\t");
				f=3;
			}
		}
		if(f==1){
			i=i+3;
		}else if(f==2){
			j=j+3;
		}else if(f==3){
			l=l+3;
		}
		num++;
			printf("\n");
	}
}

